src/Microsoft.AspNetCore.Routing/Matchers/MatcherEndpoint.cs

```diff
@@ -37,24 +37,31 @@ public sealed class MatcherEndpoint : Endpoint
             }
 
             Invoker = invoker;
-            Template = template;
-            ParsedTemlate = TemplateParser.Parse(template);
-            var mergedDefaults = GetDefaults(ParsedTemlate, new RouteValueDictionary(values));
-            Values = mergedDefaults;
             Order = order;
+
+            Template = template;
+            ParsedTemplate = TemplateParser.Parse(template);
+
+            RequiredValues = requiredValues;
+            var mergedDefaults = GetDefaults(ParsedTemplate, defaults);
+            Defaults = mergedDefaults;
         }
 
         public int Order { get; }
         public Func<RequestDelegate, RequestDelegate> Invoker { get; }
         public string Template { get; }
-        public IReadOnlyDictionary<string, object> Values { get; }
+        public IReadOnlyDictionary<string, object> Defaults { get; }
+
+        // Values required by an endpoint for it to be successfully matched on link generation
+        public RouteValueDictionary RequiredValues { get; }
```

Comment:
These should both be the same 😆 Defaults is one type and RequiredValues is another

Reply:
Oops. Good catch.